### PR TITLE
paste updated icecomm script, add code to close icecomm connection/turn ...

### PR DESCRIPTION
**Pull request by iamburitto**
I finally know what fork is for guys!

### CODE REVIEW

File: app/views/static_pages/room.html.erb [hunk 1]
```diff
@@ -57,25 +55,26 @@ var comm = new Icecomm('9gv3ODZorBS2kVuNwnugBeV432PcNX3Qn0HmW33ofxK/u5Xolu');
...
+      comm.close();
     });
+
     comm.on('data', function(options) {
     console.log("data: ",options.data);
-    //appendNewIM(options);
+    appendNewIM(options);
   });
 
-   comm.on('data',appendNewIM);
```
Comment (iamburitto) [inline app/views/static_pages/room.html.erb]: replaced line 67 - just to clean up the code, didn't need to respond to data event twice.

File: app/views/static_pages/room.html.erb [hunk 2]
```diff
@@ -57,25 +55,26 @@ var comm = new Icecomm('9gv3ODZorBS2kVuNwnugBeV432PcNX3Qn0HmW33ofxK/u5Xolu');
...
   var $inputIM    = $("#js-input-im");
   var $chatBody   = $("#js-chat-body");
-  var clientOptions;
 
   $sendButton.on('click', sendIM);
   $inputIM.on('keypress', checkEnterKey);
 
+  //With Turbolinks pages will change without a full reload, so you can't rely on the usual window.beforepageunload event. Instead Turbolinks fires events on document to provide hooks into the lifecycle of the page.
+  $(document).on("page:fetch", function(){comm.close()}); //starting to fetch a new target page
+
```
Comment (iamburitto) [inline app/views/static_pages/room.html.erb]: Lines 73-75 ensure that your local audio/vidio stream is stopped when you click on a link in rails using Turbolinks. Turbolinks fires events on document to provide hooks into the lifecycle of the page. 
page:fetch is an event indicating that we are starting to fetch a new target page from the server. documentation: https://github.com/rails/turbolinks

File: app/views/static_pages/room.html.erb [hunk 3]
```diff
@@ -57,25 +55,26 @@ var comm = new Icecomm('9gv3ODZorBS2kVuNwnugBeV432PcNX3Qn0HmW33ofxK/u5Xolu');
 
     comm.on('disconnect', function(options) {
       document.getElementById(options.callerID).remove();
-
+      comm.close();
```
Comment (iamburitto) [inline app/views/static_pages/room.html.erb]: Ensure that the local audio/vidio stream is stopped when the window is closed - closing the window fires icecomm's disconnect event.

**Comment by kmalta** [comment]
Looks good and great commenting :+1:

**Pull request merged**